openapi.yaml

```diff
@@ -0,0 +1,734 @@
+        '201':
+          description: Successfully added a game
+          schema:
+            $ref: '#/definitions/GameResult'
+        '400':
+          description: Bad request
+          schema:
+            $ref: '#/definitions/Errors'
+        '500':
+          description: Internal server error
+          schema:
+            $ref: '#/definitions/Errors'
+  /games/{gameId}:
+    parameters:
+    - $ref: '#/parameters/authorization'
+    - name: gameId
+      in: path
+      description: Id of game to return
+      required: true
+      type: string
+    get:
+      summary: Find game by ID
+      description: Returns a single game
+      operationId: getGameById
+      tags:
+        - games
+      responses:
+        '200':
+          description: Successful response
+          schema:
+            $ref: '#/definitions/GameResult'
+        '404':
+          description: Game not found
+          schema:
+            $ref: '#/definitions/Errors'
+        '500':
+          description: Internal server error
+          schema:
+            $ref: '#/definitions/Errors'
+    patch:
+      summary: Update information on a game
+      description: Pass in parameters to update fields on game resources
+      operationId: updateGameById
+      tags:
+        - games
+      parameters:
+      - name: body
+        in: body
+        required: true
+        schema:
+          $ref: '#/definitions/GamePatchBody'
+      responses:
+        '200':
+          description: Successful response
+          schema:
+            $ref: '#/definitions/GameResult'
+        '400':
+          description: Bad request
+          schema:
+            $ref: '#/definitions/Errors'
+        '404':
+          description: Game not found
+          schema:
+            $ref: '#/definitions/Errors'
+        '500':
+          description: Internal server error
+          schema:
+            $ref: '#/definitions/Errors'
+    delete:
+      summary: Delete a game
+      description: Delete a game by passing in a gameId
+      operationId: deleteGameById
+      tags:
+        - games
+      responses:
+        '204':
+          description: Game deleted successfully
+        '404':
+          description: Game not found
+          schema:
+            $ref: '#/definitions/Errors'
+        '500':
+          description: Internal server error
+          schema:
+            $ref: '#/definitions/Errors'
+  /developers:
+    parameters:
+      - $ref: '#/parameters/authorization'
+    get:
+      summary: Get complete or filtered list of developers
+      description: Retrieve complete list of developers or pass in query parameters to filter results
+      operationId: getDevelopers
+      tags:
+        - developers
+      parameters:
+        - name: name
+          in: query
+          type: string
+          description: Name of the developer
+        - name: developerId
+          in: query
+          type: string
+          description: Id of the developer
```

Comment:
What's the main difference between `GET /developers?developerId=123` and `GET /developers/123`? Do you need both of them?

Reply:
Good call. Removed `GET /developers?developerId=123`